bundles/statistics/statsgrid2016/components/IndicatorParameters.js

```diff
@@ -25,112 +34,86 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.IndicatorParameters', function
+            if (index > 0) {
+                dropdown.parent().addClass('margintop');
             }
+            cont.append(tempSelect);
+            me._selections.push(select);
+        });
 
-            // selections
-            me._selections = [];
-            indicator.selectors.forEach(function (selector, index) {
-                var placeholderText = (panelLoc.selectionValues[selector.id] && panelLoc.selectionValues[selector.id].placeholder) ? panelLoc.selectionValues[selector.id].placeholder : panelLoc.defaultPlaceholder;
-                var label = (locale.parameters[selector.id]) ? locale.parameters[selector.id] : selector.id;
-                var tempSelect = jQuery(me.__templates.select({id: selector.id, label: label}));
-                var options = {
-                    placeholder_text: placeholderText,
-                    allow_single_deselect: true,
-                    disable_search_threshold: 10,
-                    width: '100%'
-                };
-                var selections = [];
-                var select = Oskari.clazz.create('Oskari.userinterface.component.SelectList', selector.id);
-
-                selector.allowedValues.forEach(function (val) {
-                    var name = val.name || val.id || val;
-                    val.title = val.name;
-                    var optName = (panelLoc.selectionValues[selector.id] && panelLoc.selectionValues[selector.id][name]) ? panelLoc.selectionValues[selector.id][name] : name;
+        var regionSelect = me.regionSelector.create(regionsets);
+        me.regionSelector.setWidth(205);
+        // try to select the current regionset as default selection
+        regionSelect.value(me.service.getStateService().getRegionset());
+        cont.append(regionSelect.container);
 
-                    var valObject = {
-                        id: val.id || val,
-                        title: optName
-                    };
-                    selections.push(valObject);
-                });
+        me._values = {
+            ds: datasrc,
+            ind: indId,
+            regionsetComponent: regionSelect
+        };
 
-                var dropdown = select.create(selections, options);
-                dropdown.css({width: '205px'});
-                select.adjustChosen();
-                select.selectFirstValue();
-                tempSelect.find('.label').append(dropdown);
-                if (index > 0) {
-                    dropdown.parent().addClass('margintop');
-                }
-                cont.append(tempSelect);
-                me._selections.push(select);
-            });
+        me.trigger('indicator.changed', regionsets.length > 0);
+    },
+    /**
+     * @method  @public indicatorSelected  handle indicator selected
+     * @param  {Integer} datasrc indicator datasource
+     * @param  {String} indId    indicator id
+     * @param  {Object} elements elements
+     */
+    indicatorSelected: function ( datasrc, indId, elements ) {
```

Comment:
As indicatorSelected is the "go to" function in this component I would move it up/ move _createUi() down in the file. So basically indicatorSelected is called externally, which calls IndicatorParameterHandler. The IndicatorParameterHandler triggers an event that triggers a call to _createUi(). Right?